catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/metacard/MetacardApplicationTest.java

```diff
@@ -0,0 +1,192 @@
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.impl.ResultImpl;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.operation.UpdateRequest;
+import java.io.Serializable;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import javax.ws.rs.NotFoundException;
+import org.codice.ddf.catalog.ui.metacard.edit.AttributeChange;
+import org.codice.ddf.catalog.ui.metacard.edit.MetacardChanges;
+import org.codice.ddf.catalog.ui.util.EndpointUtil;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.invocation.InvocationOnMock;
+
+public class MetacardApplicationTest {
+  private static final String ID = "000000000";
+
+  private static final String TITLE_A = "Title A";
+
+  private static final String TITLE_B = "Title B";
+
+  private static final String DATE_A = "2018-12-10T13:09:40Z";
+
+  private static final String DATE_B = "2018-12-10T13:08:40Z";
+
+  private final CatalogFramework mockFramework = mock(CatalogFramework.class);
+
+  private final EndpointUtil mockUtil = mock(EndpointUtil.class);
+
+  private final MetacardApplicationUnderTest app =
+      new MetacardApplicationUnderTest(mockFramework, mockUtil);
+
+  @Test(expected = NotFoundException.class)
+  public void testPatchMetacardsWhenIdNotFound() throws Exception {
+    doReturn(Collections.emptyMap()).when(mockUtil).getMetacardsWithTagById(any(), eq("*"));
+    app.doPatchMetacards(generateTitleChange());
+  }
+
+  @Test
+  public void testPatchMetacardsWhenAttributeIsString() throws Exception {
+    ArgumentCaptor<UpdateRequest> requestCaptor = ArgumentCaptor.forClass(UpdateRequest.class);
+    when(mockFramework.update(requestCaptor.capture())).thenReturn(null);
+    doReturn(generateCatalogStateWithTitle())
+        .when(mockUtil)
+        .getMetacardsWithTagById(any(), eq("*"));
+
+    app.doPatchMetacards(generateTitleChange());
+
+    Metacard metacard = requestCaptor.getValue().getUpdates().get(0).getValue();
+    assertThat(metacard.getId(), is(ID));
+    assertThat(metacard.getTitle(), is(TITLE_B));
+  }
+
+  @Test
+  public void testPatchMetacardsWhenAttributeIsDate() throws Exception {
+    ArgumentCaptor<UpdateRequest> requestCaptor = ArgumentCaptor.forClass(UpdateRequest.class);
+    when(mockFramework.update(requestCaptor.capture())).thenReturn(null);
+    doReturn(generateCatalogStateWithCreatedDate())
+        .when(mockUtil)
+        .getMetacardsWithTagById(any(), eq("*"));
+    doAnswer(MetacardApplicationTest::doParseDate).when(mockUtil).parseDate(any());
+
+    app.doPatchMetacards(generateCreatedDateChange());
+
+    Metacard metacard = requestCaptor.getValue().getUpdates().get(0).getValue();
+    assertThat(metacard.getId(), is(ID));
+    assertThat(metacard.getCreatedDate(), is(Date.from(Instant.parse(DATE_B))));
+  }
+
+  private static List<MetacardChanges> generateTitleChange() {
```

Comment:
✏️ Consider giving each of these four `generateXxx()` methods a parameter and pass its value (instead of the constant) to the value setter. That way you don't have to look at these methods to see what the initial and updated values are for each test: you can read only the test and understand why we are expecting the expected value.

Reply:
Noted. I see how that can be unclear. I'll keep this in mind next time 👍